File: api/DesafioFeiras/desafio.feiras.domain/Feira.cs
namespace desafio.feiras.domain
{
    /// <summary>
    /// Dados que representam uma feira
    /// </summary>
    public class Feira : BaseEntity
    {
        /// <summary>
        /// Longitude da localização do estabelecimento no território do Município, conforme MDC
        /// </summary>
        public string Longitude { get; set; }
        /// <summary>
        /// Latitude da localização do estabelecimento no território do Município, conforme MDC
        /// </summary>
        public string Latitude { get; set; }
        /// <summary>
        /// Setor censitário conforme IBGE
        /// </summary>
        public string SetorCensitario { get; set; }
        /// <summary>
        /// Área de ponderação (agrupamento de setores censitários) conforme IBGE 2010
        /// </summary>
        public string AreaPonderacao { get; set; }
        /// <summary>
        /// Código do Distrito Municipal conforme IBGE
        /// </summary>
        public string CodigoDistritoIBGE { get; set; }
        /// <summary>
        /// Nome do Distrito Municipal
        /// </summary>
        public string DistritoMunicipal { get; set; }
        /// <summary>
        /// Código de cada uma das 31 Subprefeituras (2003 a 2012)
        /// </summary>
        public string CodigoSubprefeitura { get; set; }
        /// <summary>
        /// Nome da Subprefeitura (31 de 2003 até 2012)
        /// </summary>
        public string Subprefeitura { get; set; }
        /// <summary>
        /// Região conforme divisão do Município em 5 áreas
        /// </summary>
        public string RegiaoMunicipio5Areas { get; set; }
        /// <summary>
        /// Região conforme divisão do Município em 8 áreas
        /// </summary>
        public string RegiaoMunicipio8Areas { get; set; }
        /// <summary>
        /// Nome da feira livre
        /// </summary>
        public string Nome { get; set; }
        /// <summary>
        /// Número do registro da feira livre na PMSP
        /// </summary>
        public string Registro { get; set; }
        /// <summary>
        /// Nome do logradouro onde se localiza a feira livre
        /// </summary>
        public string Logradouro { get; set; }
        /// <summary>
        /// Um número do logradouro onde se localiza a feira livre
        /// </summary>
        public string NumeroLogradouro { get; set; }
        /// <summary>
        /// Bairro de localização da feira livre
        /// </summary>
        public string Bairro { get; set; }
        /// <summary>
        /// Ponto de referência da localização da feira livre
        /// </summary>
        public string PontoReferencia { get; set; }
    }
}

File: scripts/feira.py
import json

class Feira:
    longitude = ''
    latitude = ''
    setorCensitario = ''
    areaPonderacao = ''
    codigoDistritoIBGE = ''
    distritoMunicipal = ''
    codigoSubprefeitura = ''
    subprefeitura = ''
    regiaoMunicipio5Areas = ''
    regiaoMunicipio8Areas = ''
    nome = ''
    registro = ''
    logradouro = ''
    numeroLogradouro = ''
    bairro = ''
    pontoReferencia = ''    

    def __init__(self, strData):
        splittedData = strData.split(',')

        if len(splittedData) == 17:
            self.longitude = splittedData[1]
            self.latitude = splittedData[2]
            self.setorCensitario = splittedData[3]
            self.areaPonderacao = splittedData[4]
            self.codigoDistritoIBGE = splittedData[5]
            self.distritoMunicipal = splittedData[6]
            self.codigoSubprefeitura = splittedData[7]
            self.subprefeitura = splittedData[8]
            self.regiaoMunicipio5Areas = splittedData[9]
            self.regiaoMunicipio8Areas = splittedData[10]
            self.nome = splittedData[11]
            self.registro = splittedData[12]
            self.logradouro = splittedData[13]
            self.numeroLogradouro = self.__getNumber(splittedData[14])
            self.bairro = splittedData[15]
            self.pontoReferencia = splittedData[16].replace('\n','')[0:24]
        else:
            print('Tamanho de linha inválida: ', strData)

    def toJson(self):
        jsonData = json.dumps(self.__dict__)
        return jsonData

    def __getNumber(self, data):
        try:
            num = float(data)
            return str(int(num))
        except ValueError as err:
            print(err)
            return data
        
